src/goInstallTools.ts

```diff
@@ -369,21 +387,37 @@ export function installTools(missing: string[], goVersion: SemVersion): Promise<
 					resolve([...sofar, null]);
 					return;
 				}
-				const args = ['get', '-u', '-v'];
-				if (tool.endsWith('-gomod')) {
-					args.push('-d');
-				}
-				args.push(getToolImportPath(tool, goVersion));
-				cp.execFile(goRuntimePath, args, { env: envForTools }, (err, stdout, stderr) => {
-					if (stderr.indexOf('unexpected directory layout:') > -1) {
-						outputChannel.appendLine(`Installing ${tool} failed with error "unexpected directory layout". Retrying...`);
-						cp.execFile(goRuntimePath, args, { env: envForTools }, callback);
-					} else if (!err && tool.endsWith('-gomod')) {
-						const outputFile = path.join(toolsGopath, 'bin', process.platform === 'win32' ? `${tool}.exe` : tool);
-						cp.execFile(goRuntimePath, ['build', '-o', outputFile, getToolImportPath(tool, goVersion)], { env: envForTools }, callback);
-					} else {
-						callback(err, stdout, stderr);
+				// Install tools in a temporary directory, to avoid altering go.mod files.
+				fs.mkdtemp(path.join(os.tmpdir(), 'go-tools-'), (err, toolsTmpDir) => {
```

Comment:
Few more thoughts on the temp dir
- Is there a reason to have a different dir for each tool? I would imagine we create the dir once outside of the `reduce()` call and use it for all the tools
- We have a utility `getTempFilePath` that you can use which will give you a path under the temp dir that is created for this extension and is used for other purposes as well. I suggest you use this `getTempFilePath('go-tools')` to get a path and create a directory for this path using `fs.mkdirSync`
- When extension is de-activated (say when VS Code window closes or reloads), the folder would get cleaned up